sync/src/types.rs

```diff
@@ -1208,12 +1207,12 @@ impl SyncSharedState {
             // and the next time `get_block_status` would acquire the real-time
             // status via fetching block_ext from the database.
             self.remove_block_status(&block.as_ref().hash());
+            self.remove_header_view(&block.as_ref().hash());
+            self.peers()
+                .set_last_common_header(peer, block.header().clone());
         }
 
-        self.remove_header_view(&block.as_ref().hash());
-        self.peers()
-            .set_last_common_header(peer, block.header().clone());
-        ret
+        Ok(ret?)
     }
```

Comment:
Oops :eyes:

Reply:
I write `Ok(ret?)` just for convenience to convert `Result<(), ckb_error::Error>` to `Result<(), FailureError>`.
If this is a bad style, I would like to change like before, maybe more explicitly.

Reply:
I don't mean that. the original code makes me little confused.

```
if is_err {
   return 

} else {
   ~~~~;
}
~~~~;
~~~~;
```

Reply:
I don't mean that. the original code makes me little confused.

```
if is_err {
   return err;
} else {
  ...
}
...
why else
```